tests/test_exception_handling.py

```diff
@@ -0,0 +1,116 @@
+import tap_klaviyo.utils as utils_
+import unittest
+from unittest import mock
+import requests
+import singer
+
+class Mockresponse:
+    def __init__(self, resp, status_code, content=None, headers=None, raise_error=False):
```

Comment:
@harshpatel4crest You can make resp an optional argument. 
`def __init__(self, status_code, resp={}, content=None, headers=None, raise_error=False)

def klaviyo_400_error(*args, **kwargs):
    return Mockresponse(json_str, 400, raise_error=True)
`

Please follow above code for remaining functions